src/types.js

```diff
@@ -8,13 +8,25 @@ function applyRecord(columnMetadata, record) {
         case 'timestamp':
         case 'timestamptz':
           // Postgres format 2001-01-01 00:00:00 or 2001-01-01 00:00:00.123456
+          // Postgres format +292278994-08-16 23:00:00 or -292275055-05-16 23:00:00 for Infinity values
           // eslint-disable-next-line no-case-declarations
           const [, year, month, day, hour, minute, second, millisecond = ''] = record[
             column.name
           ].match(
-            /^(\d{1,4})-(\d{1,2})-(\d{1,2}) (\d{1,2}):(\d{1,2}):(\d{1,2})(?:.(\d{1,3}))?(?:\d{1,3})?$/,
+            /^([\+|\-]?\d+)-(\d{1,2})-(\d{1,2}) (\d{1,2}):(\d{1,2}):(\d{1,2})(?:.(\d{1,3}))?(?:\d{1,3})?$/,
           );
 
+          // Data Api may return timestamps out of range of JS max timestamp value of +-8640000000000000
+          // Year comparison is performed first and if the year is in range, then ISO date strings may be compared
+          const yearNumber = Number(year);
+          const isoDateString = `${year}-${month}-${day}T${hour}:${minute}:${second}.${millisecond.padEnd(3, 0)}Z`;
+          if (yearNumber > 275760 || isoDateString >= '+275760-09-13T00:00:00.000Z') {
+            return Infinity;
```

Comment:
I guess this should be assigned to the property rather than returned? I'll have a look

Reply:
Sorry for my mistake and I see you have corrected this in another PR.